### PR TITLE
Cleanup AbstractTestRest

### PR DESCRIPTION
**Pull request by dimas-b**
* Use utility method createBranch()

* Use OnBranchBuilder.branch() where appropriate

* Fix tag operations in testAssignRefToFreshMain()

Follow-up to #2522

### CODE REVIEW

File: servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractTestRest.java [hunk 1]
```diff
@@ -1840,28 +1766,20 @@ public void testAssignRefToFreshMain() throws BaseNessieClientServerException {
...
-                .reference(Branch.of(testTag, null))
-                .create()
-                .getHash())
-        .isNotNull();
-    Reference testTagRef = api.getReference().refName(testTag).get();
+    Reference testTagRef =
+        api.createReference()
+            .sourceRefName(main.getName())
+            .reference(Tag.of(testTag, main.getHash()))
+            .create();
```
[the remark this answers is not in the document]
Comment (nastra) [reply servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractTestRest.java:1779]: do you mean the `Tag.of(testTag, main.getHash())` part? My intention was to not provide a hash in the `Tag`

**Review by snazy — APPROVED** [review]

File: servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractTestRest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (dimas-b) [reply servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractTestRest.java:1779]: Yes, I mean `Tag.of`. However, a hash appears to be required for creating tags.

[the remark this answers is not in the document]
Comment (nastra) [reply servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractTestRest.java:1779]: but wasn't the code working before with `null`?

[the remark this answers is not in the document]
Comment (dimas-b) [reply servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractTestRest.java:1779]: It was using [Branch.of](https://github.com/projectnessie/nessie/pull/2550/files#diff-8a1982cdf7ac9cc4e18b6c6a9d6757c78673e09e60248d839e668e5bd5194e53L1859) not tag.

[the remark this answers is not in the document]
Comment (nastra) [reply servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractTestRest.java:1779]: ah, that was not my intention. most likely a copy/paste error. Thanks for fixing

**Review by nastra — APPROVED** [review]

**Pull request merged**